doc/developers.rst

```diff
@@ -0,0 +1,50 @@
+How to make a release
+=====================
+
+* Update the change log
+
+Use `github_changelog_generator
+<https://github.com/skywinder/github-changelog-generator#installation>`_ to
+gather all merged pull requests and closed issues during the development
+cycle. We do this because our failing discipline of writing in the
+CHANGES.rst all relevant changes, this helps our memory.
+
+.. code-block:: bash
+
+   github_changelog_generator sphinx-gallery/sphinx-gallery
+
+Read the changes in the generated CHANGELOG.md and propagate the relevant to
+CHANGES.rst
+
+* Update version
+
+Update the version in ``sphinx_gallery/__init__.py``.
+
+* Build the docs cleanly
+
+Make sure to clean all and have a clean build. Double-check visually that
+everything looks right.
+
+* Do Signed release commit
```

Comment:
I am not sure what you mean by this. Do you mean tag (you have not mentioned tag anywhere else)?

Reply:
I mean have the commit for the release be GPG signed. So that it shows in github as verified commit. I have seen many projects that sign their releases.

Regarding the tag. I don't tag the commit. I actually use the github web interphase to do the release, put the annotations and summary of the release, when you do that github puts the tag.

Reply:
Interesting, I did not know that.